drivers: ethernet: xlnx_gem: obtain data bus width at run-time, remove parameter from DT

ibirnbaum opened the pull request:
Remove the 'amba-ahb-dbus-width' DT parameter and obtain the matching value for the current target SoC at run-time during the device initialization. The matching value for the current SoC is available in the GEM register space's design_cfg5 register.

- remove 'amba-ahb-dbus-width' from the 'xlnx,gem' binding
- remove symbolic defines of valid values for use in the DT from the GEM's header file in include/dt-bindings
- remove specification of the default value from all gem[0..3] declarations in the Zynq-7000 and ZynqMP SoC DTs
- remove the corresponding field from the GEM driver's device config data struct
- remove assertion-based check of DT parameter value validity from the GEM driver
- obtain the matching value from the design_cfg5 register in eth_xlnx_gem_set_initial_nwcfg() in the GEM driver.

ibirnbaum committed:
dts: bindings: remove AMBA AHB bus width parameter from xlnx,gem binding

This parameter no longer needs to be configured at the SoC level in
the declarations of the GEM controller instances (used to require
different values for the Zynq-7000 and the ZynqMP) as the value matching
the current target SoC is now being read at run-time from a design
configuration register.

Signed-off-by: Immo Birnbaum <mail@birnbaum.immo>

ibirnbaum committed:
dts: xilinx: zynq7000: zynqmp: remove amba-ahb-dbus-width parameter

Remove the obsolete DT parameter "amba-ahb-dbus-width" from all GEM
controller instance declarations for both the Zynq-7000 and the
ZynqMP.

The value matching the current target SoC is now being read at
run-time from a design configuration register

Signed-off-by: Immo Birnbaum <mail@birnbaum.immo>

ibirnbaum committed:
drivers: ethernet: xlnx_gem: obtain AMBA AHB bus width from config register

Obtain the value for the AMBA AHB bus width (32 bit/64 bit/128 bit) from
the design_cfg5 register at init-time rather than specifying it in the
respective SoC's DT.

Signed-off-by: Immo Birnbaum <mail@birnbaum.immo>

venodela commented:
This pull request looks good to me.

I verified it on the ZCU102 ZynqMP board, where dma_bus_width_def was correctly read from designcfg_debug5 reg and written to the nw_cfg register. Ethernet communication also worked as expected.

@HariniKatakamX @michalsimek @ibirnbaum

maass-hamburg commented on dts/bindings/ethernet/xlnx,gem.yaml:
```diff
@@ -109,15 +109,6 @@ properties:
       ists within the current system setup that triggers the transmission
       of packets from within the context of the system work queue!
 
-  amba-ahb-dbus-width:
-    type: int
-    required: true
-    description: AMBA AHB data bus width.
-    enum:
-      - 0
-      - 1
-      - 2
-
```
removal needs to be mentioned in the migration guide

ibirnbaum committed:
doc: releases: migration guide entry

Added the removal of the 'amba-ahb-dbus-width' devicetree property
from the binding of the Xilinx GEM Ethernet controller driver to the
current migration guide.

Signed-off-by: Immo Birnbaum <mail@birnbaum.immo>
